monix-eval/shared/src/main/scala/monix/eval/internal/TaskShift.scala

```diff
@@ -68,9 +68,7 @@ private[eval] object TaskShift {
         })
       } catch {
         case e: RejectedExecutionException =>
-          Callback
-            .trampolined(cb)(context.scheduler)
-            .onError(e)
+          cb.onError(e)
```

Comment:
why is it no longer necessary?

Reply:
I made `Callback.trampolined()` thread-safe too and I don't want to use it just for signaling errors in such cases. But point taken, I've reintroduced the trampolined execution here.